modules/shortcodes/youtube.php

```diff
@@ -365,16 +294,159 @@ function youtube_id( $url ) {
 	return $html;
 }
 
+/**
+ * Gets the args present in the YouTube shortcode URL.
+ *
+ * @since 8.0.0
+ *
+ * @param string $url The URL of the shortcode.
+ *
+ * @return array|false The query args of the URL, or false.
+ */
+function jetpack_shortcode_youtube_args( $url ) {
+	$qargs = array();
+	if ( ! empty( $url['query'] ) ) {
+		wp_parse_str( $url['query'], $qargs );
+	} else {
+		return false;
+	}
+
+	$fargs = array();
+	if ( ! empty( $url['fragment'] ) ) {
+		wp_parse_str( $url['fragment'], $fargs );
+	}
+
+	return array_merge( $fargs, $qargs );
+}
+
 /**
  * Display the Youtube shortcode.
  *
  * @param array $atts Shortcode attributes.
+ *
+ * @return string The rendered shortcode.
  */
 function youtube_shortcode( $atts ) {
-	return youtube_id( ( isset( $atts[0] ) ) ? ltrim( $atts[0], '=' ) : shortcode_new_to_old_params( $atts ) );
+	$url = ( isset( $atts[0] ) ) ? ltrim( $atts[0], '=' ) : shortcode_new_to_old_params( $atts );
+
+	if (
+		class_exists( 'Jetpack_AMP_Support' )
+		&& Jetpack_AMP_Support::is_amp_request()
+	) {
+		return jetpack_amp_youtube_shortcode( $url );
+	} else {
+		return youtube_id( $url );
+	}
 }
 add_shortcode( 'youtube', 'youtube_shortcode' );
 
+/**
+ * Renders the [youtube] shortcode as an AMP component.
+ *
+ * @since 8.0.0
+ *
+ * @param string $url The YouTube URL.
+ *
+ * @return string The AMP-compatible rendered shortcode.
+ */
+function jetpack_amp_youtube_shortcode( $url ) {
+	$video_id = jetpack_get_youtube_id( $url );
+	if ( empty( $video_id ) ) {
+		return sprintf(
+			'<a href="%s" class="amp-wp-embed-fallback"></a>',
```

Comment:
What do you think about adding something to the link to be more descriptive? Even the URL alone may be nice I think.

```suggestion
			'<a href="%s" class="amp-wp-embed-fallback">%s</a>',
```

Reply:
That's a good idea.

Reply:
Ah, I might need to update a test for that.